lib/floe/workflow/intrinsic_functions/states/string_to_json.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    module IntrinsicFunctions
+      module States
+        class StringToJson < Floe::Workflow::IntrinsicFunction
+          def value(context, inputs)
+            arg = args.first
+            arg =
+              if arg.kind_of?(Floe::Workflow::Path) || arg.kind_of?(Floe::Workflow::IntrinsicFunction)
```

Comment:
Would it be cleaner to say `arg.respond_to?(:value)`?  Then if we have future evauable things they would just work.  That makes me wonder if it might be nice to have an Evaulable mixin, but I'm not sure what exactly we'd put into it.  I am seeing a lot of the same pattern of `if arg.start_with?("$.")` all over the place, which gets me wondering if there's way to DRY that up.